backend/Social/SocialAPI.py

```diff
@@ -45,23 +48,128 @@ def profiles():
+                        os.remove(imgFileName) # delete the existing file
+                        fd = open(imgFileName, 'wb')
+                        fd.write(imgBinary)
+                        fd.close()  
+                    else: # if file with same name does not exist
+                        fd = open(imgFileName, 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    return imgKey, imgFileLocation
+                elif imgString[:4] == "/9j/": # given image string is in base64 form and in JPG/JPEG format
+                    imgBinary = a2b_base64(imgString)
+                    imgKey = str(userID) + "/profile_pic.jpg"
+                    imgFileLocation = str(os.getcwd()) + "/profile_pic.jpg"
+                    if os.path.exists('profile_pic.jpg') == True:
+                        os.remove('profile_pic.jpg')
+                        fd = open('profile_pic.jpg', 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    else:
+                        fd = open('profile_pic.jpg', 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    return imgKey, imgFileLocation
+                elif imgString[:6] == "iVBORw": # given image string is in base64 form and in PNG format
+                    imgBinary = a2b_base64(imgString)
+                    imgKey = str(userID) + "/profile_pic.png"
+                    imgFileLocation = str(os.getcwd()) + "/profile_pic.png"
+                    if os.path.exists('profile_pic.png') == True:
+                        os.remove('profile_pic.png')
+                        fd = open('profile_pic.png', 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    else:
+                        fd = open('profile_pic.png', 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    return imgKey, imgFileLocation
+                elif imgString[:6] == "R0lGOD": # given image string is in base64 form and in GIF format
+                    imgBinary = a2b_base64(imgString)
+                    imgKey = str(userID) + "/profile_pic.gif"
+                    imgFileLocation = str(os.getcwd()) + "/profile_pic.gif"
+                    if os.path.exists('profile_pic.gif') == True:
+                        os.remove('profile_pic.gif')
+                        fd = open('profile_pic.gif', 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    else:
+                        fd = open('profile_pic.gif', 'wb')
+                        fd.write(imgBinary)
+                        fd.close()
+                    return imgKey, imgFileLocation
+                elif imgString[0] == "U": # given image string is in base64 form and in WEBP format
+                    imgBinary = a2b_base64(imgString)
```

Comment:
this line repeats itself for all cases, try to avoid repeating same code/logic whenever possible